Content.Server/Lightning/LightningSystem.cs

```diff
@@ -1,27 +1,33 @@
-﻿using System.Linq;
+using System.Linq;
 using Content.Server.Beam;
 using Content.Server.Beam.Components;
 using Content.Server.Lightning.Components;
 using Content.Shared.Lightning;
 using Robust.Server.GameObjects;
-using Robust.Shared.Physics;
-using Robust.Shared.Physics.Dynamics;
-using Robust.Shared.Physics.Events;
 using Robust.Shared.Random;
 
 namespace Content.Server.Lightning;
 
+// For maintainers (delete this text after the review)
+
+//I've redesigned the lightning system to be more optimized.
+//Previously, each lightning element, when it touched something, would try to branch into nearby entities.
+//So if a lightning bolt was 20 entities long, each one would check its surroundings and have a chance to create additional lightning...
+//which could lead to recursive creation of more and more lightning bolts and checks.
+
+//I redesigned so that lightning branches can only be created from the point where the lightning struck, no more collide checks
+//and the number of these branches is explicitly controlled in the new function.
```

Comment:
this is a good comment to leave for posterity probably, i think this is a good change though. have you checked how it affects lightning anom especially its supercrit?

Reply:
I don't know why, but my electrical anomaly didn't cause lags before the update either. But comparing the quantity and quality of lightning, it's definitely better and more concise now.